internal/health_checks/api_check.go

```diff
@@ -0,0 +1,167 @@
+	"google.golang.org/grpc/codes"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func getGCEMetadata() (resourcedetector.GCEResource, error) {
+	MetadataResource, err := resourcedetector.GetResource()
+	if err != nil {
+		return resourcedetector.GCEResource{}, fmt.Errorf("can't get resource metadata: %w", err)
+	}
+	if gceMetadata, ok := MetadataResource.(resourcedetector.GCEResource); ok {
+		return gceMetadata, nil
+	} else {
+		return resourcedetector.GCEResource{}, fmt.Errorf("not in GCE")
+	}
+}
+
+// monitoringPing reports whether the client's connection to the monitoring service and the
+// authentication configuration are valid. To accomplish this, monitoringPing writes a
+// time series point with empty values to an Ops Agent specific metric.
+// This method mirrors the "(c *Client) Ping" method in "cloud.google.com/go/logging".
+func monitoringPing(ctx context.Context, client monitoring.MetricClient, gceMetadata resourcedetector.GCEResource) error {
+	metricType := "agent.googleapis.com/agent/ops_agent/enabled_receivers"
+	now := &timestamppb.Timestamp{
+		Seconds: time.Now().Unix(),
+	}
+	value := &monitoringpb.TypedValue{
+		Value: &monitoringpb.TypedValue_Int64Value{
+			Int64Value: int64(0),
+		},
+	}
+	req := &monitoringpb.CreateTimeSeriesRequest{
+		Name: "projects/" + gceMetadata.Project,
+		TimeSeries: []*monitoringpb.TimeSeries{{
+			MetricKind: metricpb.MetricDescriptor_GAUGE,
+			ValueType:  metricpb.MetricDescriptor_INT64,
+			Metric: &metricpb.Metric{
+				Type: metricType,
+			},
+			Resource: &monitoredres.MonitoredResource{
+				Type: "gce_instance",
+				Labels: map[string]string{
+					"instance_id": gceMetadata.InstanceID,
+					"zone":        gceMetadata.Zone,
+				},
+			},
+			Points: []*monitoringpb.Point{{
+				Interval: &monitoringpb.TimeInterval{
+					StartTime: now,
+					EndTime:   now,
+				},
+				Value: value,
+			}},
+		}},
+	}
+
+	return client.CreateTimeSeries(ctx, req)
+}
+
+type APICheck struct{}
+
+func (c APICheck) Name() string {
+	return "API Check"
+}
+
+func (c APICheck) RunCheck(logger *log.Logger) error {
+	ctx := context.Background()
+	gceMetadata, err := getGCEMetadata()
+	if err != nil {
+		return fmt.Errorf("can't get GCE metadata: %w", err)
+	}
+	logger.Println(gceMetadata)
+
+	// New Logging Client
+	logClient, err := logging.NewClient(ctx, gceMetadata.Project)
+	if err != nil {
+		return err
+	}
+	logger.Printf("logging client was created successfully.")
+
+	if err := logClient.Ping(ctx); err != nil {
+		logger.Println(err)
+		var apiErr *apierror.APIError
+		if errors.As(err, &apiErr) {
+			switch apiErr.Reason() {
+			case "SERVICE_DISABLED":
+				return LOG_API_DISABLED_ERR
+			case "ACCESS_TOKEN_SCOPE_INSUFFICIENT":
+				return LOG_API_SCOPE_ERR
+			case "IAM_PERMISSION_DENIED":
+				return LOG_API_PERMISSION_ERR
+			}
+
+			switch apiErr.GRPCStatus().Code() {
+			case codes.PermissionDenied:
+				return LOG_API_PERMISSION_ERR
```

Comment:
Yes, it is possible. I added this second part of the logic, because in when getting response from the `Monitoring API`, specifically by running `monitoringPing`, the response that expected to have reason `IAM_PERMISSION_DENIED` didn't have any details and only the gRPC error code `PermissionDenied`. 

Other examples and experiments showed similar things, so decided to rely on the code if the reason wasn't set.